### PR TITLE
Add support for building ROOT with C++20

### PR DESCRIPTION
**Pull request by Axel-Naumann**
This leaves many warnings

### CODE REVIEW

File: core/foundation/inc/ROOT/RSpan.hxx [hunk 1]
```diff
@@ -14,21 +14,8 @@
...
-#elif defined(R__HAS_STD_EXPERIMENTAL_SPAN)
-
-#include <experimental/span>
-namespace std {
-  using template<class T> span = experimental::span<T>;
-
-  // TODO: using make_view() overloads
-}
-
+#ifdef __cpp_lib_span
```
Comment (hahnjo) [inline core/foundation/inc/ROOT/RSpan.hxx]: You may need to include some header to make sure this definition is visible. Can we use `__has_include` instead?

Comment (hahnjo) [reply core/foundation/inc/ROOT/RSpan.hxx]: Ah, I see you do it in a later commit. May be better to squash the two.

**Comment by Axel-Naumann** [comment]
@phsft-bot build just on mac13/cxx20,ROOT-ubuntu2204/cxx20

**Comment by Axel-Naumann** [comment]
@phsft-bot build just on mac13/cxx20, ROOT-ubuntu2204/cxx20

**Comment by vgvassilev** [comment]
FWIW, I will try seeing if I can come up with a single modulemap on this build node we use for cxx20.

**Comment by vgvassilev** [comment]
@phsft-bot build just on ROOT-ubuntu2204/cxx20

**Comment by vgvassilev** [comment]
> FWIW, I will try seeing if I can come up with a single modulemap on this build node we use for cxx20.

@Axel-Naumann, I cannot seem to log on the ubuntu node. Can you take a look?

**Comment by Axel-Naumann** [comment]
> cannot seem to log on the ubuntu node

Which one?

**Comment by vgvassilev** [comment]
> > cannot seem to log on the ubuntu node
> 
> Which one?

root-ubuntu-2204-1.cern.ch

**Comment by vgvassilev** [comment]
@phsft-bot build just on mac13/cxx20, ROOT-ubuntu2204/cxx20

**Comment by vgvassilev** [comment]
@Axel-Naumann, we seem past `std.pcm` so maybe this PR goes back to you?

**Comment by vgvassilev** [comment]
@phsft-bot build just on mac13/cxx20, ROOT-ubuntu2204/cxx20

**Comment by jalopezg-git** [comment]
@phsft-bot build just on mac13/cxx20, ROOT-ubuntu2204/cxx20

**Comment by Axel-Naumann** [comment]
This is a change too massive to be introduced into v6.28/00 at this point. We will backport it to a patch release of v6-28-00-patches.

**Review by gganis — APPROVED** [review]

**Comment by jalopezg-git** [comment]
@vgvassilev I think this needs you ;-)  I'm getting a number of errors related to cxxmodules on GCC 12.2.0/libstdc++ (see [relevant output](https://gist.github.com/jalopezg-git/383aae28d9122bc820e1668b2a2172e5)).

**Comment by vgvassilev** [comment]
@jalopezg-git, on which build node I could reproduce the failure?

**Comment by vgvassilev** [comment]
@Axel-Naumann, can we merge that in master. I don't think it regresses anything and it improves the C++20 support.

**Comment by vgvassilev** [comment]
@phsft-bot build!

**Comment by Axel-Naumann** [comment]
Failures seem spurious / pre-existing

**Comment by jalopezg-git** [comment]
> @jalopezg-git, on which build node I could reproduce the failure?

Sorry for the delay.  I was building locally on ArchLinux x86_64; I'm not sure as to whether there's a build node with gcc12.2.0.

**Pull request merged**